sources/swap/liquidity_pool.move

```diff
@@ -660,9 +676,58 @@ module liquidswap::liquidity_pool {
         exists<LiquidityPool<X, Y, Curve>>(@liquidswap_pool_account)
     }
 
-    /// Get fees (numerator, denominator).
-    public fun get_fees_config(): (u64, u64) {
-        (FEE_MULTIPLIER, FEE_SCALE)
+    /// Get fee for specific pool together with denominator (numerator, denominator).
+    public fun get_fees_config<X, Y, Curve>(): (u64, u64) acquires LiquidityPool {
+        (get_fee<X, Y, Curve>(), FEE_SCALE)
+    }
+
+    /// Get fee for specific pool.
+    public fun get_fee<X, Y, Curve>(): u64 acquires LiquidityPool {
+        assert!(coin_helper::is_sorted<X, Y>(), ERR_WRONG_PAIR_ORDERING);
+        assert!(exists<LiquidityPool<X, Y, Curve>>(@liquidswap_pool_account), ERR_POOL_DOES_NOT_EXIST);
+
+        let pool = borrow_global<LiquidityPool<X, Y, Curve>>(@liquidswap_pool_account);
+        pool.fee
+    }
+
+    /// Set fee for specific pool.
+    public entry fun set_fee<X, Y, Curve>(fee_admin: &signer, fee: u64) acquires LiquidityPool {
+        assert!(coin_helper::is_sorted<X, Y>(), ERR_WRONG_PAIR_ORDERING);
+        assert!(exists<LiquidityPool<X, Y, Curve>>(@liquidswap_pool_account), ERR_POOL_DOES_NOT_EXIST);
+        assert_pool_unlocked<X, Y, Curve>();
+        assert!(signer::address_of(fee_admin) == global_config::get_fee_admin(), ERR_NOT_ADMIN);
+
+        global_config::assert_valid_fee(fee);
+
+        let pool = borrow_global_mut<LiquidityPool<X, Y, Curve>>(@liquidswap_pool_account);
+        pool.fee = fee;
+    }
+
+    /// Get DAO fee for specific pool together with denominator (numerator, denominator).
+    public fun get_dao_fees_config<X, Y, Curve>(): (u64, u64) acquires LiquidityPool {
+        (get_dao_fee<X, Y, Curve>(), DAO_FEE_SCALE)
+    }
+
+    /// Get DAO fee for specific pool.
+    public fun get_dao_fee<X, Y, Curve>(): u64 acquires LiquidityPool {
+        assert!(coin_helper::is_sorted<X, Y>(), ERR_WRONG_PAIR_ORDERING);
+        assert!(exists<LiquidityPool<X, Y, Curve>>(@liquidswap_pool_account), ERR_POOL_DOES_NOT_EXIST);
+
+        let pool = borrow_global<LiquidityPool<X, Y, Curve>>(@liquidswap_pool_account);
+        pool.dao_fee
+    }
+
+    /// Set DAO fee for specific pool.
+    public entry fun set_dao_fee<X, Y, Curve>(fee_admin: &signer, dao_fee: u64) acquires LiquidityPool {
+        assert!(coin_helper::is_sorted<X, Y>(), ERR_WRONG_PAIR_ORDERING);
+        assert!(exists<LiquidityPool<X, Y, Curve>>(@liquidswap_pool_account), ERR_POOL_DOES_NOT_EXIST);
+        assert_pool_unlocked<X, Y, Curve>();
+        assert!(signer::address_of(fee_admin) == global_config::get_fee_admin(), ERR_NOT_ADMIN);
```

Comment:
Is it correct that the fee admin can set the dao fee and not the dao admin?

Reply:
Correct, dao only withdraws rewards for now.

Reply:
@borispovod